stdlib/public/Cxx/CxxSet.swift

```diff
@@ -94,4 +103,18 @@ extension CxxUniqueSet {
     let inserted: Bool = insertionResult.second
     return (inserted, rawIterator.pointee)
   }
+
+  /// Removes the given element from the set.
+  ///
+  /// - Parameter member: An element to remove from the set.
+  @discardableResult
+  @inlinable public mutating func remove(_ member: Element) -> Element? {
+    let iter = self.__findMutatingUnsafe(member)
+    guard iter != __endMutatingUnsafe() else {
+      return nil
+    }
+    let value = iter.pointee
+    __eraseUnsafe(iter)
+    return value
```

Comment:
I think it should be possible to just call `self.erase(member)` if you add `func erase` method to the protocol. The original C++ `erase` method has an overload that should be usable from Swift without needing the unsafe underscored methods:

(4) at https://en.cppreference.com/w/cpp/container/set/erase

Reply:
@egorzhdan  Thank you for your response.   Here are some of my thoughts, please correct me if I am wrong:

To return the actual removed element, it is necessary to perform the `__findUnsafe` operation. Using `__eraseUnsafe` leverages the result of `__findUnsafe`. If we directly use func `erase`, it would involve two lookup operations due to both `erase` and `__findUnsafe` performing a query.

Reply:
Ah, you're correct, we would need to use the unsafe overload to get the removed element back.